plugins/simple_methods_anc/src/openjdk/jcov/filter/simplemethods/EmptyMethods.java

```diff
@@ -24,28 +24,19 @@
  */
 package openjdk.jcov.filter.simplemethods;
 
-import org.objectweb.asm.tree.ClassNode;
-import org.objectweb.asm.tree.MethodNode;
-
+import java.lang.classfile.ClassModel;
+import java.lang.classfile.MethodModel;
+import java.lang.classfile.Opcode;
 import java.util.function.BiPredicate;
 
-import static org.objectweb.asm.Opcodes.RETURN;
+import static openjdk.jcov.filter.simplemethods.Utils.isSimpleInstruction;
 
-public class EmptyMethods implements BiPredicate<ClassNode, MethodNode> {
+public class EmptyMethods implements BiPredicate<ClassModel, MethodModel> {
     @Override
-    public boolean test(ClassNode node, MethodNode m) {
-        int index = 0;
-        int opCode = -1;
-        //skip all instructions allowed to get values
-        for(; index < m.instructions.size(); index++) {
-            opCode = m.instructions.get(index).getOpcode();
-            if(opCode >=0) {
-                if (!Utils.isSimpleInstruction(opCode)) {
-                    break;
-                }
-            }
-        }
-        //that should be a return
-        return opCode == RETURN;
+    public boolean test(ClassModel node, MethodModel m) {
+        if (m.code().isPresent()) {
+            var next = new InstructionIterator(m.code().get()).next(i -> !isSimpleInstruction(i.opcode()));
```

Comment:
Bad practice to use var in "shared" projects except tests - a human isn't javac, for readability it would be much better to see a type of the result.

Reply:
I am happy to follow any coding guidelines. Was there an update to the openjdk guidelines in regards to the new language features? This code uses multiple new language features, I do not immediately see why "var" should be an exception.

Reply:
I am however updating all the cases where the type is non-obvious - that point is IMO valid.

Reply:
Any strong opinion on the remaining vars?

Reply:
there is just one thing: code should be readable without a strong understanding of what a method is returning in an assignment statement. If the receiver does not follow naming conventions, then at least its type should be shown in the code.

Reply:
I agree fully, and I have changed the code to that end. The only vars left are where it is obvious what the type is.